app/routes.py

```diff
@@ -1,2 +1,272 @@
+        return check_not_found
+
+    task = Task.query.get(task_id)
+    request_body = request.get_json()
+    task.title = request_body["title"]
+    task.description = request_body["description"]
+    db.session.commit()
+
+    response_body = {}
+    response_body["task"] = task.to_dict()
+    return jsonify(response_body), 200
+
+
+@tasks_bp.route("/<task_id>", methods=["PATCH"])
+def update_task_parameter(task_id):
+    check_not_found = is_parameter_found(Task, task_id)
+    if check_not_found:
+        return check_not_found
+
+    task = Task.query.get(task_id)
+    request_body = request.get_json()
+    if "title" in request_body:
+        task.name = request_body["title"]
+    if "description" in request_body:
+        task.description = request_body["description"]
+    
+    db.session.commit()
+    return make_response(f"Task {task_id} successfully updated!", 200)
+
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def mark_task_complete(task_id):
+    check_not_found = is_parameter_found(Task, task_id)
+    if check_not_found:
+        return check_not_found
+
+    task = Task.query.get(task_id)
+    task.completed_at = datetime.now(timezone.utc)
+    response_body = {}
+    response_body = task.to_dict()
+
+    db.session.commit()
+    slack_url = "https://slack.com/api/chat.postMessage"
+    slack_bot_data = {"token": os.environ.get("ADA_BOT_TOKEN"),
+                        "channel": "task-notifications", 
+                        "text": (f"Someone just completed the task {task.title}") 
+                        }
+    requests.post(slack_url, slack_bot_data)
+    
+
+    return jsonify({"task": response_body}), 200
+
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_task_incomplete(task_id):
+    check_not_found = is_parameter_found(Task, task_id)
+    if check_not_found:
+        return check_not_found
+
+    task = Task.query.get(task_id)
+    task.completed_at = None
+    response_body = {}
+    response_body = task.to_dict()
+
+    db.session.commit()
+    return jsonify({"task": response_body}), 200
+
+@tasks_bp.route("/<task_id>", methods=["DELETE"])
+def delete_task(task_id):
+    check_not_found = is_parameter_found(Task, task_id)
+    if check_not_found:
+        return check_not_found
+
+    task = Task.query.get(task_id)
+    task_title = task.title
+    response_str = f'Task {task_id} "{task_title}" successfully deleted'
+
+    db.session.delete(task)
+    db.session.commit()
+    return jsonify({
+        "details": response_str
+        }), 200
+
+
+
+#GOAL ROUTES 
+
+@goals_bp.route("", methods=["POST"])
+def create_goal():
+    request_body = request.get_json()
+    if "title" not in request_body:
+        return jsonify({"details": "Invalid data"}), 400
+
+    new_goal = Goal(title=request_body["title"],
+                    )
+
+    db.session.add(new_goal)
+    db.session.commit()
+    response_body = {}
+    response_body["goal"] = new_goal.to_dict()
```

Comment:
As above, we do not need an empty dictionary.
```suggestion
    response_body["goal"] = new_goal.to_dict()
```